packages/api/src/useFetch.ts

```diff
@@ -8,12 +9,25 @@ import type {
     TSocketResponseData,
 } from '../types';
 
+const AUTH_REQUIRED_ENDPOINTS = ['balance'];
+
 const useFetch = <T extends TSocketEndpointNames>(name: T, ...props: TSocketAcceptableProps<T, true>) => {
     const prop = props?.[0];
     const payload = prop && 'payload' in prop ? (prop.payload as TSocketRequestPayload<T>) : undefined;
     const options = prop && 'options' in prop ? (prop.options as TSocketRequestQueryOptions<T>) : undefined;
 
-    return useQuery<TSocketResponseData<T>, unknown>(getQueryKeys(name, payload), () => send(name, payload), options);
+    let enabled = true;
+
+    if (AUTH_REQUIRED_ENDPOINTS.includes(name)) {
+        const state = queryClient.getQueryState<TSocketResponseData<'authorize'>>(['authorize']);
+
+        enabled = Boolean(state?.data?.authorize);
+    }
```

Comment:
@heorhi-deriv Does this work? 👀 The code I sent was just an example I didn't check it 👀

Reply:
@farzin-deriv yes, it works. balance request is going after authorize. there are no logouts also, as i faced before